aplib.net-demo/Assets/Scripts/Inventory/EquipmentInventory.cs

```diff
@@ -0,0 +1,154 @@
+    /// </remarks>
+    /// <value>
+    /// <c>true</c> if switching to the new item is enabled; otherwise, <c>false</c>.
+    /// </value>
+    /// <seealso cref="EquipmentInventory.EquipItem(IEquipment)" />
+    /// <seealso cref="EquipmentInventory.CurrentEquipment" />
+    public bool SwitchToNewItem;
+
+    /// <summary>
+    /// The default equipment collection.
+    /// </summary>
+    /// <remarks>
+    /// This collection contains the default equipment items that are added to the equipment inventory when it is instantiated.
+    /// </remarks>
+    [field: SerializeField]
+    public List<Equipment> DefaultEquipment { get; set; } = new();
+
+    private int _currentEquipmentIndex;
+
+    private List<Equipment> _equipmentList;
+
+    /// <summary>
+    /// Specifies whether the EquipmentInventory has any items.
+    /// </summary>
+    /// <remarks>
+    /// This property returns <c>true</c> if the EquipmentInventory has at least one item in its EquipmentList, and
+    /// <c>false</c> otherwise.
+    /// </remarks>
+    /// <value><c>true</c> if the EquipmentInventory has items; otherwise, <c>false</c>.</value>
+    /// <seealso cref="EquipmentInventory.EquipItem(IEquipment)" />
+    public bool HasItems => _equipmentList?.Count > 0;
+
+    /// <summary>
+    /// Gets the currently equipped item from the equipment inventory.
+    /// </summary>
+    /// <value>
+    /// The currently equipped item.
+    /// </value>
+    public Equipment CurrentEquipment => _equipmentList[_currentEquipmentIndex];
+
+    private void Start()
+    {
+        // Link to the input manager
+        InputManager.Instance.equipmentInventory = this;
+
+        // If the default equipment is empty, add the default equipment to the equipment list.
+        // This is to ensure backward compatibility with the previous implementation.
+        if (DefaultEquipment?.Count <= 0)
+        {
+            Debug.LogWarning(
+                "Default equipment is empty! Rolling back to obsolete implementation. \n "
+                + "Make sure to set the default equipment in the inspector."
+            );
+            GameObject playerObject = GameObject.FindGameObjectWithTag("Player");
+
+            if (playerObject)
+            {
+                DefaultEquipment = new List<Equipment>(playerObject.transform.GetComponentsInChildren<Equipment>());
+            }
+        }
+
+        _equipmentList = DefaultEquipment.ToList();
+
+        // Set every equipment item to be inactive, except the first one.
+        for (int i = 1; i < _equipmentList.Count; i++)
+        {
+            _equipmentList[i].gameObject.SetActive(false);
+        }
+    }
+
+    /// <summary>
+    /// Equip the specified equipment item to the EquipmentInventory.
+    /// </summary>
+    /// <param name="equipment">The equipment item to be equipped.</param>
+    public void EquipItem(Weapon equipment)
```

Comment:
```suggestion
    public void EquipItem(Equipment equipment)
```
Zou dit ook niet Equipment moeten zijn of kom je dan verkeerd uit met types?